src/Metadata/MetadataBase.php

```diff
@@ -253,4 +254,41 @@ protected function ensureIsTrusted(bool $allowUntrustedAccess = false): void
             throw new \RuntimeException("Cannot use untrusted '{$this->getRole()}'. metadata.");
         }
     }
+
+    /**
+     * Checks for a rollback attack.
+     *
+     * Verifies that an incoming remote version of a metadata file is greater
+     * than or equal to the last known version.
+     *
+     * @param \Tuf\Metadata\MetadataBase $remoteMetadata
+     *     The latest metadata fetched from the remote repository.
+     * @param integer|null $expectedRemoteVersion
+     *     If not null this is expected version of remote metadata.
+     *
+     * @return void
+     *
+     * @throws \Tuf\Exception\PotentialAttackException\RollbackAttackException
+     *     Thrown if a potential rollback attack is detected.
+     */
+    public function checkRollbackAttack(MetadataBase $remoteMetadata, int $expectedRemoteVersion = null): void
```

Comment:
Since this now on `MetadataBase` I am wondering if we should not remove the `$expectedRemoteVersion` parameter here.

The only time we pass it is in `updateRoot()` so it only applies for the Root metadata. We could override this in `RootMetadata` and add the `$expectedRemoteVersion` parameter there